api/wikis/serializers.py

```diff
@@ -7,8 +7,27 @@
 
 from framework.auth.core import Auth
 
+
 class WikiSerializer(JSONAPISerializer):
 
+    def to_representation(self, data, envelope='data'):
+        if data.node.is_registration:
+            serializer = RegistrationWikiSerializer(data, context=self.context)
```

Comment:
Good catch. In this case we aren't leaking data between the two endpoints, but I've added [tests](https://github.com/CenterForOpenScience/osf.io/pull/6380/commits/f8e7e935fc78d683bb6fba0eb61caa783bd51295#diff-93502c9a04d7a3055b84e71a66469bfeR140) for confirmation anyways.